src/runtime/pkg/govmm/qemu/qemu.go

```diff
@@ -2471,14 +2471,26 @@ const (
 	Unix QMPSocketType = "unix"
 )
 
-// QMPSocket represents a qemu QMP socket configuration.
+const (
+	// Socket using a human-friendly text-based protocol.
+	Hmp string = "hmp"
+
+	// Socket using a richer json-based protocol.
+	Qmp string = "qmp"
+
+	// Same as Qmp with pretty json formatting.
+	QmpPretty string = "qmp-pretty"
+)
+
+// QMPSocket represents a qemu QMP or HMP socket configuration.
 // nolint: govet
 type QMPSocket struct {
```

Comment:
Just as something to consider, now that `QMPSocket` is not necessarily QMP, a more general name, perhaps `ControlSocket` (?), might be more appropriate.

Reply:
You're right. I thought about doing that but maybe not in this PR because the renaming will be quite noisy and I really want this fix to reach main and stable-3.2 as soon as possible.

Reply:
Sure, doing the rename in a subsequent PR would be perfectly fine.